python/ray/tests/test_runtime_env_profiler.py

```diff
@@ -0,0 +1,207 @@
+import os
+import glob
+import sys
+
+import pytest
+
+import ray
+from ray._private.runtime_env.nsight import parse_nsight_config
+from ray._private.test_utils import wait_for_condition
+from ray.exceptions import RuntimeEnvSetupError
+
+
+def wait_for_report(logs_dir, num_reports):
+    assert len(os.listdir(logs_dir)) == num_reports
+    return True
+
+
+class TestNsightProfiler:
+    @pytest.mark.skipif(
+        os.environ.get("CI") and sys.platform != "linux",
+        reason="Requires PR wheels built in CI, so only run on linux CI machines.",
+    )
+    def test_nsight_basic(
+        self,
+        shutdown_only,
+    ):
+        """Test Nsight profile generate report on logs dir"""
+        ray.init()
+        session_dir = ray.worker._global_node.get_session_dir_path()
+        logs_dir = f"{session_dir}/logs/nsight"
+
+        # test nsight default config
+        @ray.remote(runtime_env={"nsight": "default"})
+        def test_generate_report():
+            return 0
+
+        ray.get(test_generate_report.remote())
+
+        wait_for_condition(wait_for_report, logs_dir=logs_dir, num_reports=1)
+        nsys_reports = glob.glob(os.path.join(f"{logs_dir}/*.nsys-rep"))
+        assert len(nsys_reports) == 1
+        for report in nsys_reports:
+            assert "worker_process_" in report
+            os.remove(report)
+
+    @pytest.mark.skipif(
+        os.environ.get("CI") and sys.platform != "linux",
+        reason="Requires PR wheels built in CI, so only run on linux CI machines.",
+    )
+    def test_nsight_custom_name(
```

Comment:
```suggestion
    def test_nsight_custom_option(
```
(since the purpose of the test is to make sure options work, not the custom name. There's no reason to test if custom name works because that's already should've tested with nsight itself)